### PR TITLE
feat: Implement Avail validium data attestation contracts

### PR DESCRIPTION
**Pull request by QEDK**
# What ❔
This PR adds data attestation support for validiums using Avail's bridge.

## Why ❔
This adds verification and storage of data attestations onchain allowing for provable data availability on the settlement layer.

## Checklist
- [X] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [X] Documentation comments have been added / updated.

### CODE REVIEW

File: l2-contracts/src/deploy-l2-da-validator-on-l2-through-l1.ts [hunk 1]
```diff
@@ -81,7 +81,7 @@ async function main() {
       let l2DaValidatorAddress;
       let l1DaValidatorAddress;
       if (cmd.validiumMode) {
-        l2DaValidatorAddress = await deployContractOnL2ThroughL1(deployer, "LayerNameL2DAValidator", chainId, gasPrice);
+        l2DaValidatorAddress = await deployContractOnL2ThroughL1(deployer, "AvailL2DAValidator", chainId, gasPrice);
```
[the remark this answers is not in the document]
Comment (QEDK) [reply l2-contracts/src/deploy-l2-da-validator-on-l2-through-l1.ts:84]: I'm not immediately clear on changes are required - should I write the implementation to read and write bytecode for the Avail contracts?

[the remark this answers is not in the document]
Comment (QEDK) [reply l2-contracts/src/deploy-l2-da-validator-on-l2-through-l1.ts:84]: This is addressed partially - unable to get the artifact for the `AvailL1DAValidator` contract in the `state-transition` folder

**Comment by dimazhornyk** [comment]
Closing this one because these changes were already added to [sync-layer-stable](https://github.com/matter-labs/era-contracts/tree/sync-layer-stable)

**Comment by durimb** [comment]
era beqaraj

**Pull request closed without merging**